### PR TITLE
Add FXIOS-6508 [v117] ShareExtensionCoordinator flag feature.

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6508)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14602)

### Description

Add ShareExtensionCoordinator flag feature to Nimbus.

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

File: nimbus-features/shareExtensionCoordinatorRefactor.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+          Enables the feature
+        type: Boolean
+        default: false
+    defaults:
+      - channel: beta
+        value:
+          enabled: false
+      - channel: developer
+        value:
+          enabled: true
```
Comment (OrlaM) [inline nimbus-features/shareExtensionCoordinatorRefactor.yaml:18]: This should be disabled in the developer channel until the feature is complete.

**Review by OrlaM — CHANGES_REQUESTED** [review]

File: nimbus-features/shareExtensionCoordinatorRefactor.yaml (hunk 1 above)
Comment (FilippoZazzeroni) [reply nimbus-features/shareExtensionCoordinatorRefactor.yaml]: Ok perfect i modified it right now

**Review by OrlaM — APPROVED** [review]

**Comment by FilippoZazzeroni** [comment]
@OrlaM  i see the swiftlint pipeline failed, should i rebase on main?  it failed on file i didn't modify

**Comment by OrlaM** [comment]
Looks like the latest version of swiftint found some rules violations it previously missed. I'll create a PR to address them and let you know when it's merged so you can rebase.

**Comment by FilippoZazzeroni** [comment]
Ok perfect thank you

**Comment by lmarceau** [comment]
@FilippoZazzeroni I think you can rebase and it should be good now

**Comment by FilippoZazzeroni** [comment]
ok perfect i'll do it now

**Pull request merged**